packages/web-components/fast-element/src/binding/signal.ts

```diff
@@ -14,7 +14,16 @@ const subscribers: Record<
     undefined | Subscriber | Set<Subscriber>
 > = Object.create(null);
 
+/**
+ * The gateway to signal APIs.
+ * @public
+ */
 export const Signal = Object.freeze({
```

Comment:
Wait are you using Signals in FAST?!  <img src="https://media.giphy.com/media/l46Cc1X4M6IH5c45G/giphy.gif"/>

Reply:
This is a different type of signal than what is being talked about in the blogs these days. HOWEVER, FAST has been using what the blogs are talking about as "signals" since day one. Our reactivity system is exactly that. We have fine-grained updates based on observable changes, and computed properties automatically track their dependencies. Example:

```ts
export class Person {
  @observable firstName: string;
  @observable lastName: string;

  get fullName() {
    return `${this.firstName} ${this.lastName}`;
  }
}

const template = html<Person>`
  Full Name: ${x => x.fullName}
`;

const model = new Person();
template.render(model, document.body);

model.firstName = "Rob";
// and now the exact text node where fullname is rendered is surgically updated with the new value computed from first and last name.
```

We've been operating this way since the prototype of FAST in January, 2020. Our reactivity system and our templating system are both independent from Web Components, so you can use them on their own.

The "signal" that is referenced in this PR, allows you to specifically name a templating binding and then tell the engine exactly when you want that to update. Consider that you want to display a "time since" based on the date, and you want to update all "time since" representations across the entire app every 60 seconds. You can set those up as "signal bindings" and then send the signal whenever you want them to update.

```ts
const template = html`
  Posted ${signal(x => timeSince(x.postedAt), "time-since")}.
`;

setTimeout(() => Signal.send("time-since"), 60 * 1000);

// Every 60 seconds all bindings labeled as "time-since" will be re-evaluated and their associated DOM will be surgically updated in place.
```

Reply:
Aha... I am corrected!